core/txpool/legacypool/legacypool.go

```diff
@@ -575,7 +575,7 @@ func (pool *LegacyPool) Pending(filter txpool.PendingFilter) map[common.Address]
 		// If the miner requests tip enforcement, cap the lists now
 		if minTipBig != nil && !pool.locals.contains(addr) {
 			for i, tx := range txs {
-				if tx.EffectiveGasTipIntCmp(minTipBig, pool.priced.urgent.GetNativeBaseFee()) < 0 {
+				if tx.EffectiveGasTipIntCmp(minTipBig, pool.priced.urgent.GetBaseFeeIn(tx.FeeCurrency())) < 0 {
```

Comment:
Hy @karlb I don't understand this. The minBigTip is surely provided in the native currency, so how can comparing it to the tip in the fee currency be reasonable?

Reply:
Passing in the base fee in the fee currency is necessary to make the tip calculation work. Otherwise, a wrong tip would be calculated due to mixing a native base fee with feecurrency based maxPriorityFee and gasFeeCap.

But you are right that afterwards, a fee currency tip is compared to a native minTipBig and that is wrong. Since `minTipBig` is 1 by default and we don't change that anywhere this won't cause any noticable problems, but it is still wrong.

Reply:
```suggestion
				// Here we compare the minBigTip in native currency with the tip calculated in the
				// fee currency, this is wrong but required in order to make the tip calculation work.
				// The tip is configured via miner.Config.GasPrice which by default is set to 1 so shouldn't
				// cause any problems. TODO disable the flag to configure miner.Config.GasPrice ?																 				
				if tx.EffectiveGasTipIntCmp(minTipBig, pool.priced.urgent.GetBaseFeeIn(tx.FeeCurrency())) < 0 {
```

Reply:
Thanks for the clarification @karlb, should we disable the flag to configure the tip then?

Reply:
If we do this correctly, people sending a single fee currency wei as maxPriorityFee will not get their txs included if their currency is worth less than Celo. That is totally correct, I just wonder if we might break some txs that wrongly accepted before.

Reply:
Fixed in https://github.com/celo-org/op-geth/pull/244/commits/3cc1577c27a55eae86a33792b5e4fbd208ffcc3f

Reply:
Nice!